### PR TITLE
[CORE] Various fixes/improvements for unit test

### PR DESCRIPTION
**Pull request by smuzaffar**
Updated unit tests so that they can be run from their own separate directory. Mostly 
- convert `<bin ...>` to `<test....>`
- use full path e.g `CMSSW_BASE/src/...` instead of `./src/`

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: IOPool/Common/test/BuildFile.xml [hunk 1]
```diff
@@ -1,14 +1,3 @@
-<environment>
-  <bin file="TestIOPoolCommon.cpp" name="TestEdmFastMerge">
```
Comment (makortel) [inline IOPool/Common/test/BuildFile.xml:2]: Could the `TestIOPoolCommon.cpp` be removed as well?

Comment (smuzaffar) [reply IOPool/Common/test/BuildFile.xml:2]: yes

**Comment by smuzaffar** [comment]
please test

**Comment by makortel** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**